### PR TITLE
UCT/IB: Implement multipath for RoCE LAG and IB LMC

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 19-Feb-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/4233edc9226e0ebf1d1914a466a54a22f7f41b6b/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/6f56ddf81c6b2c703b91c5e880f6e96ed0506230/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/bc31f40fe7104512115e7df04dc4153c82160788/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/5b74b4e2ae749e68e97b597ba46f7484282e085f/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/4d93ee13b98b6d3c24cf5f4c074a1f3af05ace50/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/6cc7945a28378841cc5a6fb09c02adcc06cab136/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/986e0cefdc18e0e0c1672cc34703bc11f648a0f2/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/a6376368172903124d6634ca130693d85d97b3ea/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/3346c483794a0aa672e328920b6dd125cbf05b36/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/c69017614c1c35cfe779c3118b946758fde02a59/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/147a452a77a84c129430ab6f95cc5cdb4df1e272/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/15c378e739fdc9d2de3d200ad4933004fa5d16f6/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/e77697c6a9a61e32d3440136e053fbbaf0029346/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/eb39c2d2f99e5730a9bf84d01cab8f9c010c88b1/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/590eb2e7e503ad2a7a2b00380635172013fbeda2/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/7beea4debe493635c6a60996cd66bbeafa30c0d6/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/152ce6c456d0b169cd91b2279d642dcdc508958a/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/32554f539b2f8fe7384304af1af796d020f32e30/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/96d2effd7aa7aaa2a622924a0c322049c70fa033/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/152bd1bb153d1f28313b42a5f2e65a30d8386350/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/1da98c5c46dc3ff1c538f121717f24956145ba10/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/cf5cde9c2b0b4d73f33853aa57e4844f4bf41cc2/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/508444d85fd434c735f7897ee0fa3d5c7fd504f1/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/4e38ed774141c973624b62d253b27ec395049304/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6ce89df68ce93f40432c092f743c9644/raw/e8c42fbf807358e28e41315356f92f83c8877c3c/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -813,6 +840,25 @@ static ucs_status_t uct_ib_iface_set_moderation(struct ibv_cq *cq,
...
+        if (uct_ib_iface_is_roce(iface)) {
+            /* RoCE - number of paths is RoCE LAG level */
+            iface->num_paths = uct_ib_device_get_roce_lag_level(dev);
+        } else {
+            /* IB - number of paths is LMC level */
+            ucs_assert(iface->path_bits_count > 0);
+            iface->num_paths = iface->path_bits_count;
+        }
+    } else {
+        iface->num_paths = config->num_paths;
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: do we need to check for wrong values?

File: src/uct/ib/base/ib_iface.c [hunk 2]
```diff
@@ -156,7 +156,20 @@ ucs_config_field_t uct_ib_iface_config_table[] = {
...
 
-  {"LID_PATH_BITS", "0-17",
+  {"NUM_PATHS", "auto",
+   "Number of paths to expose for the interface. 'auto' means use "
+   UCS_PP_MAKE_STRING(UCT_IB_DEV_MAX_PORTS) " paths\n"
+   "for RoCE LAG devices and 2^LMC paths for InfiniBand fabric. In some\n"
+   "cases using multiple paths can get higher bandwidth overall.",
+   ucs_offsetof(uct_ib_iface_config_t, num_paths), UCS_CONFIG_TYPE_ULUNITS},
+
+  {"ROCE_PATH_FACTOR", "1",
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: why it needs to be configurable?

File: src/uct/ib/base/ib_iface.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/base/ib_iface.c]: we use modulo so if num_paths>path_bits_count it wraps around

File: src/uct/ib/base/ib_iface.c (hunk 2 above)
Comment (yosefe) [reply src/uct/ib/base/ib_iface.c]: the value we want to use can depend on fabric switch model/configuration/wiring in case of RoCE

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 24-Feb-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/c668c40fd050e64e21664a2eb4aa36357eb8264b/_main) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/d1425347694fe2bd365a0d555a77ee2476dd1961/hpc-arm-cavium-jenkins_W0) | :question: ABORTED
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/426e5b7f3f18c167bd05af66abc6bf6e57bdddee/hpc-arm-cavium-jenkins_W2) | :question: ABORTED
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/2524abd8ba01a2d1b53236704530de6fbb8ace03/hpc-arm-cavium-jenkins_W3) | :question: ABORTED
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/6d4f7f4385502dc64f68641a3ba4f88ec9381d05/hpc-arm-hwi-jenkins_W0) | :question: ABORTED
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/e41c0d536bf9ee6bfcf2f7acb3b11988aa8d9eb8/hpc-arm-hwi-jenkins_W1) | :question: ABORTED
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/5defd3dab1622a60e21b823d7da495f4ebbac5bc/hpc-arm-hwi-jenkins_W2) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/8cbc1cb370cae935e697e4de7760ffb268f11aff/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/8057e93166a4f14f61d610fcf5ae0d5044851a09/r-vmb-ppc-jenkins_W0) | :question: ABORTED
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/84832c5a95faedba290ed38e9d332698a848a873/r-vmb-ppc-jenkins_W1) | :question: ABORTED
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/a8b3cf78e52e68b29204fc6a17f8bc835a50d957/r-vmb-ppc-jenkins_W2) | :question: ABORTED
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/3031a6143fc23b2c9eaf5cd281c6f05401d8e532/r-vmb-ppc-jenkins_W3) | :question: ABORTED
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/1cd273bc14b08bd1a2efb432efa12dead2ea7308/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/3978ac285db22bdd73fc5345ed3a83a58fa24c22/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/c8cbf5a5d09efa452e2d611ea1acb6d71a936f53/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/2819748126c62b4673483d18c87fb90d34a19f5c/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/1a62f1c923df5f528a46be13aeb6faf32e23eee6/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/02d27c8a788ca527023819a9317345aeb6886595/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/7f5abbad1ddb664986e683e3b9f57731f1f8cd28/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/d5a3a29e3c675830dad0662dcb548b5c5b378bd9/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/d1cfa5941f4271921709ee42af0ed7d39b90ffca/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/ec620bca2b0a28fcc3c375ecdc9a7d7350b4a893/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/3ab16fd48dd6cc301d008af5ce9478d1e01cbf78/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/b35ec0edaac4f4d72f366c072fd8a07af50e5a94/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/e130fc6aa7c7dde1aab1dd7183702aaa/raw/dcba2428cce521b0656268e45e23a871eae80289/hpc-test-node-new_W3) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 24-Feb-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/5bd75a27917238311e92826e1080025ed29e7f35/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/946ccd3421f3df64ef48ad77037e69e0fdb05b65/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/345e47c7992eb4ba3e0f350677f87436d66209a7/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/708366bf2c5dfd1f3eb7fdbc369bdeeab58e1717/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/c076b7bd65308d68772e15967853d7a827d5ea44/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/ff0c5c225d75adfd28a1fe0ced3d2293b5e8b687/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/5a04c10ea185e5e4071c6510976cb5c03a1b4869/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/8aaded053c0070877eecf9682d8dc4cb1df8177c/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/751821f74a3921f63f1aac80203084f51b7c16a9/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/c1142140da247f25a2cddcaa1da0b4ece39c26a6/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/53cc64461be81f2dccdc198d338fc9f894a0d6fb/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/0322e1a1cd2b959bcd81552b80f1389fc0e9cc99/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/ab25ad4a466edc0141438594019d098ec41e601b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/fc086be3d9d0960c1e4a2af98e9b62312989d51d/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/4526cd3b6a4c7b5d04f75e6ce6b5c3c8457bd415/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/767ab30e0de4d04c2c6120247538daa8f7011db7/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/99de05fb32e77593a99aaf7efa9ee1cb3636263d/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/0ad99603921ae9fcccffbdded776f4895d6659c0/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/cb23040933d517441a05e3c0a2c2cb9c97ea2b07/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/21a5014fffbedb1e6a459d945c9f6accfdd2a784/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/a120a5bbef5ef3531e7cde2ce87c5415edc6ac69/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/cde2e442fa8faa1c3b33ebbcc2b2d98159af6519/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/5e74967da0e0b476a1126ad7e828d267fab657b2/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/0455bdaa0c88599b4252741127d5215fbab6083b/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/cffae259d1798cb644cbb2179c5ff544/raw/f3cbb3b4c04e5ca636b4abcf2e4c064bbd7025d4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@brminich @alinask can you pls take a look?

File: src/uct/ib/base/ib_device.h [hunk 3]
```diff
@@ -46,6 +46,7 @@
 #define UCT_IB_SITE_LOCAL_PREFIX    be64toh(0xfec0000000000000ul) /* IBTA 4.1.1 12b */
 #define UCT_IB_SITE_LOCAL_MASK      be64toh(0xffffffffffff0000ul) /* IBTA 4.1.1 12b */
 #define UCT_IB_DEFAULT_ROCEV2_DSCP  106  /* Default DSCP for RoCE v2 */
+#define UCT_IB_ROCE_UDP_SPORT_BASE  0xC000
```
Comment (alinask) [inline src/uct/ib/base/ib_device.h]: imho SRC_PORT is clearer

File: src/uct/ib/base/ib_iface.c [hunk 4]
```diff
@@ -157,7 +157,20 @@ ucs_config_field_t uct_ib_iface_config_table[] = {
    "IB Hop limit / RoCEv2 Time to Live. Should be between 0 and 255.\n",
    ucs_offsetof(uct_ib_iface_config_t, hop_limit), UCS_CONFIG_TYPE_UINT},
 
-  {"LID_PATH_BITS", "0-17",
+  {"NUM_PATHS", "auto",
+   "Number of paths to expose for the interface. 'auto' means use "
```
Comment (alinask) [inline src/uct/ib/base/ib_iface.c]: maybe explain what interface is in this case? does the user know what it means here?

**Comment by yosefe** [comment]
@alinask the documentation now looks like this:
```
#
# Number of connections that should be created between a pair of communicating
# endpoints for optimal performance. The default value 'auto' behaves according
# to the port link layer:
#  RoCE       - 2 for LAG port, otherwise - 1.
#  InfiniBand - As the number of path bits enabled by fabric's LMC value and selected
#               by UCX_IB_LID_PATH_BITS configuration.
#
# syntax:    unsigned long: <number>, "inf", or "auto"
# inherits:  UCX_RC_NUM_PATHS, UCX_IB_NUM_PATHS
#
UCX_RC_VERBS_NUM_PATHS=auto
```

File: src/uct/ib/rdmacm/rdmacm_md.c [hunk 5]
```diff
@@ -242,7 +242,7 @@ uct_component_t uct_rdmacm_component = {
     .name               = "rdmacm",
     .md_config          = {
         .name           = "RDMA-CM memory domain",
-        .prefix         =  "IB_",
+        .prefix         = "RDMACM_",
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_md.c]: fixed copy/paste in rdmacm

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 02-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/27df27fea5e4801effef701f4bb10ee84d24766f/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/17a59a64a20abac4f9c9aaccc77e4f12e92792fd/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/602076158be6a8f42be8803b65fe2fff0942460f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/d9e86f2fe4dfd649ca100ab18533a3913cff48cf/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/eba06e5f23388a17e03a890a8d483a933b8425cd/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/b5fbb76db36714556f979eaeb52234b86f38582f/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/705918cc29bd04c2c34e92b1b72d16838d6677e2/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/9125a605665114368d425904608ed45741d9b0d1/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/ade64b0ad08060e760f79f2a391f8e4538afb9a2/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/63a54b60367aaaa8e98c52859a35ec888ed2906a/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/ed6169590b70a59b33e2052c960ee78b6bca5829/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/c60b0d638e0fe3c3095d0ff095d3093608699fb4/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/1aead64d7b696ca2533ed2672a515f4a7fdfb28d/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/082b3b2be3a7f9dfa125378ef44c49e7b3245206/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/f1d33f2ed3c49f43d5d2542f1c991708a5634ea8/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/f06ad64d22cffe553e82e46c31a7f6e0ffd04478/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/107b6f8e99e695d88165ef7708bd0cd256163117/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/92b6027291ac35ffa79e81f29f3cbe55505ad31a/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/f9f427666703623f650e351eca65c0f8b5af9de3/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/6b0be303239d73d1ce43841abb07e137c6d001f2/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/66ce300beaa408431f9654769ac3f2a1e9d5d718/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/f2d67bd103783c206250c3bf9199080aebd01a79/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/817beb052f560ca7c52dac4d223a8f30507b4a6b/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/418763116a60fdb269657b70029576a5ce89fdb9/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/78830973f1e01ce4d042ea9771e85bc8/raw/f9727329e8e9a8cc414de4dfeb96d4f99625f4fb/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**